src/hooks/preparse.ts

```diff
@@ -0,0 +1,99 @@
+  if (!argv.includes('--flags-dir') || !options.flags?.['flags-dir']) return argv;
+
+  const { default: fs } = await import('node:fs/promises');
+  const { default: path } = await import('node:path');
+  context.debug('Initial argv', argv.join(' '));
+  const flagsToIgnore = new Set(
+    Object.entries(options.flags ?? {})
+      .filter(
+        ([, flagOptions]) =>
+          // don't ignore if flag can take multiple values
+          (flagOptions.type === 'option' && flagOptions.multiple !== true) || flagOptions.type === 'boolean'
+      )
+      .filter(
+        ([flagName, flagOptions]) =>
+          // ignore if short char flag is present
+          argv.includes(`-${flagOptions.char}`) ||
+          // ignore if long flag is present
+          argv.includes(`--${flagName}`) ||
+          // ignore if --no- flag is present
+          (flagOptions.type === 'boolean' && flagOptions.allowNo && argv.includes(`--no-${flagName}`))
+      )
+      .flatMap(([flagName, flagOptions]) => {
+        // Also ignore the --no- flag if boolean flag allows it
+        if (flagOptions.type === 'boolean' && flagOptions.allowNo) {
+          return [flagName, `no-${flagName}`];
+        }
+
+        return [flagName];
+      })
+  );
+  context.debug('Flags to ignore', flagsToIgnore);
+
+  const flagsDir = argv[argv.indexOf('--flags-dir') + 1];
+  if (!flagsDir) {
+    context.debug('No flags dir provided');
+    return argv;
+  }
+  if (flagsDir.startsWith('-')) {
+    context.debug(`No flags dir provided, got ${flagsDir}`);
+    return argv;
+  }
+
+  async function safeReadDir(filePath: string): Promise<string[]> {
+    try {
+      return await fs.readdir(filePath);
+    } catch (err) {
+      context.debug('No flags dir found');
+      context.debug(err);
+      return [];
+    }
+  }
+
+  async function safeReadFile(filePath: string): Promise<string | undefined> {
+    try {
+      return await fs.readFile(filePath, 'utf8');
+    } catch (err) {
+      context.debug(filePath, err);
+    }
+  }
+
+  const filesInDir = await safeReadDir(flagsDir);
+  context.debug('Files in dir', filesInDir);
+  const flagsToInsert = await Promise.all(
+    filesInDir
+      // ignore files that were provided as flags
+      .filter((f) => !flagsToIgnore.has(f))
+      .map(async (file) => {
+        const { name, ext } = path.parse(file);
+        const contents = await safeReadFile(path.join(flagsDir, file));
+        if (contents === undefined) {
+          return [name, undefined] satisfies [string, undefined];
+        }
+        const values = ext === '.json' ? [JSON.stringify(JSON.parse(contents))] : contents?.trim().split('\n');
+        return [name, values] satisfies [string, string[]];
+      })
+  );
+  context.debug('Flags to insert', flagsToInsert);
+  for (const [flag, values] of flagsToInsert) {
+    for (const value of values ?? []) {
+      argv.push(flag.length === 1 ? `-${flag}` : `--${flag}`);
```

Comment:
let's generally avoid mutating passed-in inputs.  I saw oclif is doing some [...spreads] and is going to reassign `this.argv` with the return value so it probably works ok.